3 - PB/Documentazione esterna/Specifica Tecnica/Specifica Tecnica.typ

```diff
@@ -146,7 +147,308 @@ Ciò avviene prestando attenzione alle tecnologie impiegate sia nel front-end ch
+
+=== Strategy
+Consente di definire una famiglia di algoritmi, incapsularli in classi separate e renderli intercambiabili. In questo modo è possibile applicare l'algoritmo appropriato senza dover conoscere i dettagli implementativi.
+
+
+
+== Classi e Componenti
+Ciascun layer possiede il suo indipendente sistema di classi e componenti e prevede metodi per comunicare con i layer adiacenti.
+
+=== Persistence layer
+Mediante delle Server Action offerte da Next.js, vengono eseguite delle query SQL atte alla lettura dei dati utili all'applicazione da un database esterno.
+Esse sono implementate e rese disponibili in file separati, organizzati nell'omonima cartella "Server Action", contenente:
+
+- *getAllBins*: ritorna le informazioni di tutti i bin lette dal database;
+
+- *getBinById*: dato un codice identificativo univoco, ritorna le informazioni relative al bin corrispondente lette dal database;
+
+- *getAllProducts*: ritorna le informazioni di tutti i prodotti lette dal database;
+
+- *getProductById*: dato un codice identificativo univoco, ritorna le informazioni relative al prodotto corrispondente lette dal database;
+
+- *SVGSanitize*: dato un path ad un file SVG caricato, ritorna il contenuto del relativo file SVG sanificato, ovvero normalizzato e reso sicuro, prevenendo attacchi XSS;
+
+- *readSavedSVG*: ritorna il contenuto del file SVG salvato su server;
+
+- *saveSVG*: dato il contenuto di un file SVG, esso viene salvato come _saved.svg_;
+
+- *getAllZones*: ritorna le informazioni di tutte le zone lette dal database;
+
+- *getBinsByZoneId*: dato un codice identificativo univoco, ritorna le informazioni relative a tutti i bin contenuti nella zona corrispondente lette dal database;
+
+- *getZoneById*: dato un codice identificativo univoco, ritorna le informazioni relative alla zona corrispondente lette dal database.
+
+
+Al fine di agevolare la divisione tra il Persistence layer ed il Business layer, viene utilizzato il pattern Repository mediante classi che implementano l'interfaccia `dataRepositoryInterface`, quali:
+
+- *binRepository*: è responsabile dell'ottenimento dei dati relativi agli oggetti `Bin`;
+
+- *productRepository*: è responsabile dell'ottenimento dei dati relativi agli oggetti `Product`;
+
+- *zoneRepository*: è responsabile dell'ottenimento dei dati relativi agli oggetti `Zone`.
+
+Il pattern Repository impiega in maniera consequenziale le classi correlate del pattern Data Mapper, le quali implementano l'interfaccia `DataMapperInterface`.
+Di seguito sono elencate le classi specifiche:
+
+- *binMapper*: è responsabile della creazione di oggetti `Bin`;
+
+- *productMapper*: è responsabile della creazione di oggetti `Product`;
+
+- *zoneMapper*: è responsabile della creazione di oggetti `Zone`.
+
+
+
+Diagramma delle classi del layer di persistenza: #TODO
```

Comment:
Non è una richiesta di correzione ma un dubbio mio: probabilmente (e anche sicuramente) mi sono perso io qualcosa, ma come mai non ci sono più le immagini?

Reply:
perchè quelle che che erano presenti erano ancora work in progress e pensavo di aggiungerle in una task successiva ma sono quasi pronte quelle corrette quindi possono anche venire aggiunte in questa pr